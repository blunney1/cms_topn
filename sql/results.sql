--
--Check functions of the extension which returns results like frequencies and the top n:
--topn, cms_topn_info, cms_topn_frequency.
--

--topn
--check with null and empty values
SELECT topn(NULL, NULL::integer);
SELECT topn(cms_topn(2), NULL::text);

--check with wrong parameters
SELECT topn(cms_topn_add(cms_topn(2),2),NULL::text);

--cms_topn_info
--check with null and empty values
SELECT cms_topn_info(NULL);
SELECT cms_topn_info(cms_topn(1));
SELECT cms_topn_info(cms_topn(2, 0.01, 0.99));

--check normal cases
SELECT cms_topn_info(cms_topn_add(cms_topn(1), 5));
SELECT cms_topn_info(cms_topn_add(cms_topn(2), 5));
SELECT cms_topn_info(cms_topn_add(cms_topn(2, 0.1, 0.9), '5'::text));

--cms_topn_frequency
--check with null and empty values
SELECT cms_topn_frequency(NULL, NULL::text);
SELECT cms_topn_frequency(NULL, 3::integer);
SELECT cms_topn_frequency(cms_topn_add(cms_topn(1),5), NULL::text);
SELECT cms_topn_frequency(cms_topn(3), NULL::bigint);

--check normal cases
CREATE TABLE results (
	cms_topn_column cms_topn
);

INSERT INTO results(cms_topn_column) SELECT cms_topn_add_agg(int_column, 3) FROM numbers;
SELECT cms_topn_frequency(cms_topn_column, 0) FROM results;
SELECT cms_topn_frequency(cms_topn_column, 1) FROM results;
SELECT cms_topn_frequency(cms_topn_column, 2) FROM results;
SELECT cms_topn_frequency(cms_topn_column, 3) FROM results;
SELECT cms_topn_frequency(cms_topn_column, 4) FROM results;
SELECT cms_topn_frequency(cms_topn_column, 5) FROM results;
SELECT cms_topn_frequency(cms_topn_column, -1) FROM results;
SELECT cms_topn_frequency(cms_topn_column, NULL::integer) FROM results;

DELETE FROM results;
INSERT INTO results(cms_topn_column) SELECT cms_topn_add_agg(text_column, 2) FROM strings;
SELECT cms_topn_frequency(cms_topn_column, '0'::text) FROM results;
SELECT cms_topn_frequency(cms_topn_column, '1'::text) FROM results;
SELECT cms_topn_frequency(cms_topn_column, '2'::text) FROM results;
SELECT cms_topn_frequency(cms_topn_column, '3'::text) FROM results;
SELECT cms_topn_frequency(cms_topn_column, '4'::text) FROM results;
SELECT cms_topn_frequency(cms_topn_column, '5'::text) FROM results;
SELECT cms_topn_frequency(cms_topn_column, '6'::text) FROM results;
SELECT cms_topn_frequency(cms_topn_column, NULL::text) FROM results;




