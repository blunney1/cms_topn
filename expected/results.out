--
--Check functions of the extension which returns results like frequencies and the top n:
--topn, cms_topn_info, cms_topn_frequency.
--
--topn
--check with null and empty values
SELECT topn(NULL, NULL::integer);
 topn 
------
 
(1 row)

SELECT topn(cms_topn(2), NULL::text);
ERROR:  there is not any items in the cms_topn
--check with wrong parameters
SELECT topn(cms_topn_add(cms_topn(2),2),NULL::text);
ERROR:  not proper cms_topn for the result type
--cms_topn_info
--check with null and empty values
SELECT cms_topn_info(NULL);
 cms_topn_info 
---------------
 
(1 row)

SELECT cms_topn_info(cms_topn(1));
                    cms_topn_info                    
-----------------------------------------------------
 Sketch depth = 5, Sketch width = 2719, Size = 106kB
(1 row)

SELECT cms_topn_info(cms_topn(2, 0.01, 0.99));
                   cms_topn_info                   
---------------------------------------------------
 Sketch depth = 5, Sketch width = 272, Size = 10kB
(1 row)

--check normal cases
SELECT cms_topn_info(cms_topn_add(cms_topn(1), 5));
                    cms_topn_info                    
-----------------------------------------------------
 Sketch depth = 5, Sketch width = 2719, Size = 106kB
(1 row)

SELECT cms_topn_info(cms_topn_add(cms_topn(2), 5));
                    cms_topn_info                    
-----------------------------------------------------
 Sketch depth = 5, Sketch width = 2719, Size = 106kB
(1 row)

SELECT cms_topn_info(cms_topn_add(cms_topn(2, 0.1, 0.9), '5'::text));
                  cms_topn_info                  
-------------------------------------------------
 Sketch depth = 3, Sketch width = 28, Size = 0kB
(1 row)

--cms_topn_frequency
--check with null and empty values
SELECT cms_topn_frequency(NULL, NULL::text);
 cms_topn_frequency 
--------------------
                   
(1 row)

SELECT cms_topn_frequency(NULL, 3::integer);
 cms_topn_frequency 
--------------------
                   
(1 row)

SELECT cms_topn_frequency(cms_topn_add(cms_topn(1),5), NULL::text);
 cms_topn_frequency 
--------------------
                   
(1 row)

SELECT cms_topn_frequency(cms_topn(3), NULL::bigint);
 cms_topn_frequency 
--------------------
                   
(1 row)

--check normal cases
CREATE TABLE results (
	cms_topn_column cms_topn
);
INSERT INTO results(cms_topn_column) SELECT cms_topn_add_agg(int_column, 3) FROM numbers;
SELECT cms_topn_frequency(cms_topn_column, 0) FROM results;
 cms_topn_frequency 
--------------------
                200
(1 row)

SELECT cms_topn_frequency(cms_topn_column, 1) FROM results;
 cms_topn_frequency 
--------------------
                 10
(1 row)

SELECT cms_topn_frequency(cms_topn_column, 2) FROM results;
 cms_topn_frequency 
--------------------
                150
(1 row)

SELECT cms_topn_frequency(cms_topn_column, 3) FROM results;
 cms_topn_frequency 
--------------------
                  3
(1 row)

SELECT cms_topn_frequency(cms_topn_column, 4) FROM results;
 cms_topn_frequency 
--------------------
              40000
(1 row)

SELECT cms_topn_frequency(cms_topn_column, 5) FROM results;
 cms_topn_frequency 
--------------------
               6000
(1 row)

SELECT cms_topn_frequency(cms_topn_column, -1) FROM results;
 cms_topn_frequency 
--------------------
                  0
(1 row)

SELECT cms_topn_frequency(cms_topn_column, NULL::integer) FROM results;
 cms_topn_frequency 
--------------------
                   
(1 row)

DELETE FROM results;
INSERT INTO results(cms_topn_column) SELECT cms_topn_add_agg(text_column, 2) FROM strings;
SELECT cms_topn_frequency(cms_topn_column, '0'::text) FROM results;
 cms_topn_frequency 
--------------------
                  2
(1 row)

SELECT cms_topn_frequency(cms_topn_column, '1'::text) FROM results;
 cms_topn_frequency 
--------------------
                  1
(1 row)

SELECT cms_topn_frequency(cms_topn_column, '2'::text) FROM results;
 cms_topn_frequency 
--------------------
              15000
(1 row)

SELECT cms_topn_frequency(cms_topn_column, '3'::text) FROM results;
 cms_topn_frequency 
--------------------
                 20
(1 row)

SELECT cms_topn_frequency(cms_topn_column, '4'::text) FROM results;
 cms_topn_frequency 
--------------------
                  6
(1 row)

SELECT cms_topn_frequency(cms_topn_column, '5'::text) FROM results;
 cms_topn_frequency 
--------------------
              70000
(1 row)

SELECT cms_topn_frequency(cms_topn_column, '6'::text) FROM results;
 cms_topn_frequency 
--------------------
                  0
(1 row)

SELECT cms_topn_frequency(cms_topn_column, NULL::text) FROM results;
 cms_topn_frequency 
--------------------
                   
(1 row)

